### ISSUE TITLE
Sensor isn't updated after reboot until a new value is read

### ISSUE THREAD

**Issue by jyavenard** [gtdiehl/iotawatt_ha#15]
This is an issue I discussed in https://community.home-assistant.io/t/custom-component-iotawatt-energy-monitor-integration/254110/107?u=jyavenard

Last night I updated the HA version just past midnight. 
![iota_integration](https://user-images.githubusercontent.com/505748/130374354-a320e0a2-153c-4728-8350-670baf26851d.jpg)

What we can see when dragging the mouse over that graph, that the last value for the Export Output was just at the time of the reboot at 12:03:24.
Following that there's been no update to the Export sensor until 9:00:39 where it reads -188.1W

Somehow, after a reboot, the sensor isn't updated until it reads a different value from the last (or non zero I'm not sure which condition is most correct)

As I use the Riemann Sum integration with the right method, it considers that -188.1W is the average wattage since the last read.
so 0.1881kWh * 9h = 1.06kWh will be what the first value stored will be for the Export Energy for the day; skewing the result of the entire day.

After a restart, the sensor must be continuously updated, not only when a new, different value is read.
Otherwise calculating the energy used during the interval is going to provide the wrong answer.

**Comment by OzGav** [comment]
I agree. Note that a new integration called powercalc https://github.com/bramstroker/homeassistant-powercalc does just this. It updates every 10 minutes or at a different interval as specified by the option `scan_interval`

**Comment by jyavenard** [comment]
I believe the root cause of this is #16 

I should have amended that bug maybe rather than creating a new one.

> Note that a new integration called powercalc https://github.com/bramstroker/homeassistant-powercalc

the issue here is the original value of the sensor created by the iotawatt integration ; if the sensor isn't updated, it doesn't matter how regularly you integrate it. you'll always get the wrong result
